src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -571,3 +636,147 @@ function mapAndFilterNotices(notices: GraphQlNotice[]): Notice[] {
 
   return filterNotices(mappedNotices);
 }
+
+export function setFilterSegments(transportModes: GraphQlTransportModes[]) {
+  return [
+    {
+      filters: [{ select: [{ transportModes: transportModes }] }],
+    },
+    {
+      filters: [{ select: [{ transportModes: transportModes }] }],
+    },
+  ];
+}
+
+export function findTripPatternCombinationsList(
+  tripPatternCombinations: ViaTripsWithDetailsQuery['viaTrip']['tripPatternCombinations'],
+): { from: number; to: number }[][] {
+  return tripPatternCombinations.map((combinations) =>
+    combinations.map((combination) => ({
+      from: combination.from!,
+      to: combination.to!,
+    })),
+  );
+}
+
+export function findTripPatternsFromViaTo(
+  tripPatternCombinations: { from: number; to: number }[][],
+  tripPatternsFromVia: ViaTripsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'],
```

Comment:
Is there no separate type for these kind of trip patterns?

Reply:
The types ViaTripsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'] and ViaTripsWithDeatilsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'] comes from via-trip-generated.ts and via-trips-with-details-generated.ts. It does not exist any type  of separate types for these trip patterns.